### PR TITLE
Add the base of SendMail action.

### PR DESCRIPTION
**Pull request by hosszukalman**

### CODE REVIEW

File: src/Plugin/Action/SendMail.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ *       label = @Translation("To"),
+ *       description = @Translation("The e-mail address or addresses where the message will be sent to. The formatting of this string must comply with RFC 2822."),
+ *       multiple = TRUE
+ *     ),
+ *     "subject" = @ContextDefinition("string",
+ *       label = @Translation("Subject"),
+ *       description = @Translation("The mail's subject.")
+ *     ),
+ *     "message" = @ContextDefinition("string",
+ *       label = @Translation("Message"),
```
Comment (fago) [inline src/Plugin/Action/SendMail.php]: Let's add a todo for having a long text widge for message also.

File: src/Plugin/Action/SendMail.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+use Drupal\Core\Mail\MailManager;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Send mail' action.
+ *
+ * @Action(
+ *   id = "rules_send_mail",
```
Comment (fago) [inline src/Plugin/Action/SendMail.php]: Should be rules_mail_send

File: src/Plugin/Action/SendMail.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add group information from Drupal 7.
+ * @todo: Language context has "options list", "default value" and "default
+ *   mode" settings in Drupal 7.
+ * @todo Get RulesPlugin $element variable in execute function to generate
+ *   unique key for the mail. Now $this->getPluginId() and $this->getBaseId() is
+ *   used, but it's needs to be fixed.
+ */
+class SendMail extends RulesActionBase implements ContainerFactoryPluginInterface {
```
Comment (fago) [inline src/Plugin/Action/SendMail.php]: Class should be SystemMailSend.

File: src/Plugin/Action/SendMail.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    $reply = $this->getContextValue('reply');
+    $language = $this->getContextValue('language');
+    $langcode = isset($language) ? $language->getId() : LanguageInterface::LANGCODE_NOT_SPECIFIED;
+    $params = array(
+      'subject' => $this->getContextValue('subject'),
+      'message' => $this->getContextValue('message'),
+      'langcode' => $langcode,
+    );
+    // Set a unique key for this mail.
+    $name = $this->getBaseId() ? $this->getBaseId() : 'unnamed';
```
Comment (fago) [inline src/Plugin/Action/SendMail.php]: Should have a todo to figure out a unique mail key.

Comment (hosszukalman) [inline src/Plugin/Action/SendMail.php]: It is marked in https://github.com/fago/rules/pull/100/files#diff-86c0528b6ffdc72b552f2304a2b6fd50R55

**Pull request merged**